index.js

```diff
@@ -6,29 +6,97 @@ const db = require('./data/db.js');
 
 const server = express();
 
-// add your server code starting here
+server.use(express.json());
+const cors = require('cors')
+// server.use(cors());
 
-server.get('/', (req, res) =>{
+// add your server code starting here
+//get posts
+server.get('/api/posts',cors(), (req, res) =>{
     db.find().then(posts => {
-        res.status(418).json(posts);
+        res.status(200).json(posts);
     }).catch(err => {
-        res.json(err);
+        res.status(500).json({ error: "The posts information could not be retrieved." });
     })
     
 });
 
+// get post by id
 server.get('/api/posts/:id', (req, res) => {
     const id = req.params.id;
 
     db.findById(id).then(posts => {
         if(posts){
             res.status(200).json(posts)
         }else{
-            res.status(404).json({message: 'post not found'})
+            res.status(404).json({ message: "The post with the specified ID does not exist." })
         }
         
     })
-    .catch(err => res.status(500).json(err))
+    .catch(err => res.status(500).json({ error: "The post information could not be retrieved." }))
 });
 
-server.listen(5000, () => console.log('server running'));
\ No newline at end of file
+server.post('/api/posts', (req, res) => {
+    const postInfo = req.body;
+    console.log('body:', req.body);
+    if((postInfo.title !== undefined)&& (postInfo.contents !== undefined)){
```

Comment:
It's probably be worth it to destructure these to make this a little shorter and more readable (instead of something that people like me will gloss over when trying to get up to speed).